### PR TITLE
remove obsolete instructions on hacking clang+gfortran

### PR DESCRIPTION
**Pull request by davydden**
in reference to https://github.com/LLNL/spack/pull/2192#issuecomment-257427916

### CODE REVIEW

**Comment by citibeth** [comment]
So what is the best practice for MacOS users now?  And can it be done without involving Homebrew/MacPorts/etc?

**Comment by davydden** [comment]
@citibeth just set up paths to 
```
f77: /path/to/bin/gfortran
fc: /path/to/bin/gfortran
```
should be enough with https://github.com/LLNL/spack/pull/2192. Essentially https://github.com/LLNL/spack/pull/2192 did the hacking.

**Comment by tgamblin** [comment]
I think @citibeth is looking for step-by-step instructions to tell a user where they can get `gfortran` for their mac in the first place -- basically the getting started guide is supposed to tell them how to bootstrap, so this doc section should probably say something like "you will need fortran compilers, which you can either install with `spack` (`spack install gcc`) or with `homebrew` (`brew install gcc`)".  Then tell them how they can type `spack compiler add` to get it into `compilers.yaml`.

**Comment by citibeth** [comment]
Complete instructions would explain you have to run `spack install gcc`
first.  Then you have to locate where `gfortran` is in what you just
built.  Then you can set up the paths...

**Comment by adamjstewart** [comment]
Also, very common problem, we'll have users run Spack and later discover that they are missing Fortran compilers. They will install gfortran and run `spack compiler find` and continue on their merry way, only to find that Spack didn't add anything to the existing compilers. Maybe this is a bug that should get fixed? Otherwise, we need to document that you actually have to manually add these.

**Comment by tgamblin** [comment]
@davydden: willing to take a stab at that?

**Comment by davydden** [comment]
@tgamblin @citibeth @adamjstewart : I added detailed instructions on how to get Clang+gfortran on macOS. Have a look.

**Comment by citibeth** [comment]
All the information is there now.  But can it be a bit more organized?  Something like below.  The idea is to give step-by-step instructions.  I've suggested separating out instructions for Spack vs. non-Spack, assuming the non-Spack procedure is pretty similar for the three non-Spack options.  Setting up with Spack is a little different.

We should also consider other Homebrew-like systems; specifically MacPorts and Fink.  I think Fink especially should be considered because I believe it is a binary installer, rather than building from source.  That would probably be more convenient in this case than building GCC yourself, and quite appropriate IMHO.

Setting Up Clang+GFortran on macOS
=========================

You can use Spack, Homebrew, MacPorts or Fink to set up

Setting up with Spack
-----------------------
1. Run spack install gcc
2. Use spack location -i
3. Edit files, etc....

Setting up with non-Spack
---------------------------
1. Use either:
    a) brew install gcc
    b) port install gcc
    c) fink install gcc
2. Locate using `which gfortran`
3. Edit files, etc...

**Comment by davydden** [comment]
@citibeth I don't think we need to have more levels for this doc. Currently it is already 3rd level:
`Getting started -> Compiler configuration -> Mixed Toolchains`. Since `gclang+gfortran` is only a single example of mxied toolchains, then we would need to have at least two more levels: `Setting Up Clang+GFortran on macOS` + `GCC via Spack/ GCC from Homebrew/Fink/etc`. IMHO it is way to many.

**Comment by davydden** [comment]
I will try to organize it more using bullet points...

**Comment by citibeth** [comment]
>
> I will try to organize it more using bullet points...
>

Good idea; my suggestion was not intended to suggest specific additional
levels, just a general way the information could be organized.

**Comment by davydden** [comment]
3 steps now: install / locate clang / edit. Since I can not visualize it, let's see what Travis has to say...

**Comment by adamjstewart** [comment]
> Since I can not visualize it, let's see what Travis has to say...

There's two easy ways to visualize it:

1. [Build it yourself](http://spack.readthedocs.io/en/latest/contribution_guide.html#documentation-tests) and open the resulting HTML files in your web browser. Kind of a pain in the ass, but tells you exactly what it will look like.

2. On GitHub, under Files changed, click on View at the top right. This allows you to view the whole file, rendering the .rst on the fly. This is way more convenient, but doesn't support inter-file links or macros like `.. note::`

**Comment by davydden** [comment]
@adamjstewart thanks! I checked it on Github, looks reasonable to me.

**Comment by davydden** [comment]
i did various modifications and i think i should stop editing this, at least for a while. Let me know if there are some grammar issues or alike to be fixed.

**Comment by citibeth** [comment]
I got my Mac set up to visualize pretty easily with MacPorts: `port install
sphinx` or something.

**Comment by alalazo** [comment]
@davydden If you build it yourself while you work on it you can remove the `-E` option in the `Makefile` to avoid rebuilding everything every time.

**Comment by davydden** [comment]
I don't edit documentation much, so I try to avoid installing neither `macports` nor `homebrew` :smile: p.s. i used  both in the past 4-5 years.

**Comment by tgamblin** [comment]
@davydden I applaud your dogfooding!

**Comment by citibeth** [comment]
The reason I used MacPorts... I run mostly a Python3 environment.  Since
Spack is Python2, its Sphinx needs to be running Python2.

No problem, so I just have to build `py-sphinx^python@2.7`.  However... for
some strange reason, Python2 won't build for me; see #1100 .  Rather than
digging into the problem, I just installed with MacPorts and moved on.

**Comment by davydden** [comment]
@tgamblin given the lack of further comments on this doc update, do you want to merge it?

File: lib/spack/docs/getting_started.rst [hunk 1]
```diff
@@ -414,7 +414,17 @@ provides no Fortran compilers.  The user is therefore forced to use a
 mixed toolchain: XCode-provided Clang for C/C++ and GNU ``gfortran`` for
 Fortran.
 
-In the simplest case, you can just edit ``compilers.yaml``:
+#. First, you need to make sure that command-line tools are installed. To that
+   end run ``$ xcode-select --install``.
+
+#. Next you need to run ``$ spack compiler find`` to locate Clang.
+
```
Comment (citibeth) [inline lib/spack/docs/getting_started.rst]: The words "First" and "Next" aren't need above; they are implied by the numbered list.

**Comment by davydden** [comment]
I removed `First` and `Next`.

**Comment by tgamblin** [comment]
@citibeth: homebrew installs precompiled binaries ("Kegs") and AFAIK not many people use fink anymore.

**Pull request merged**